auction-backend/src/main/java/com/atlantbh/internship/auction/app/service/impl/DidYouMeanServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.atlantbh.internship.auction.app.service.impl;
+
+import com.atlantbh.internship.auction.app.dto.search.SearchSuggestion;
+import com.atlantbh.internship.auction.app.model.impl.JaccardIndex;
+import com.atlantbh.internship.auction.app.repository.ItemRepository;
+import com.atlantbh.internship.auction.app.service.DidYouMeanService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class DidYouMeanServiceImpl implements DidYouMeanService {
+    private final ItemRepository itemRepository;
+
+    public DidYouMeanServiceImpl(final ItemRepository itemRepository) {
+        this.itemRepository = itemRepository;
+    }
+
+    @Override
+    public SearchSuggestion didYouMean(final String input) {
+        if (input.isEmpty()) {
+            return new SearchSuggestion("");
+        }
+
+        List<String> items = itemRepository.getListOfItemNames();
```

Comment:
If we have millions of items, there is a potential for a memory overflow problem. However, for internship purposes, we don't need to address this issue. Nevertheless, it's important to be aware of it. If you were to transition the implementation to the frontend, you could reuse pagination you already have.

Reply:
final